java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -572,7 +572,23 @@ public void createIndex(
       boolean replace) {
     try (LockManager.ReadLock readLock = lockManager.acquireReadLock()) {
       Preconditions.checkArgument(nativeDatasetHandle != 0, "Dataset is closed");
-      nativeCreateIndex(columns, indexType.getValue(), name, params, replace);
+      boolean isVectorIndex =
+          indexType == IndexType.VECTOR
+              || indexType == IndexType.IVF_FLAT
+              || indexType == IndexType.IVF_SQ
+              || indexType == IndexType.IVF_PQ
+              || indexType == IndexType.IVF_HNSW_SQ
+              || indexType == IndexType.IVF_HNSW_PQ;
+
+      if (isVectorIndex) {
```

Comment:
the params should always be non-null, even for scalar index we will initialize an empty one